Add arrangement management api page (#29)
wso2/docs-ob-toolkit-cds: en/docs/try-out/arrangement-management-api-flow.md

> When a customer revoke granted consents there should be a mechanism to inform relevant parties that the particular CDR 

Reviewer: ```suggestion
When a customer revokes a granted consent there should be a mechanism to inform relevant parties that the particular CDR 
```

> When a customer revokes a granted consent there should be a mechanism to inform relevant parties that the particular CDR 

Reviewer: ```suggestion
# CDR Arrangement Management API

When a customer revokes a granted consent there should be a mechanism to inform relevant parties that the particular CDR 
```

Arrangement ID is not valid anymore. The Data Recipients use the CDR Arrangement Management API, which facilitates 
this requirement. If this communication does not take place, the Data Holder will continue to expose the customer's 
data and the Data Recipient will continue to have the customer's data within their system. Therefore, it is important 
to communicate the revocation to both parties to protect customer data and prevent misuse.

> This page explains how to configure and deploy the CDR Arrangement Management API as per the latest changes introduced

Reviewer: ```suggestion
This page explains how to configure and deploy the CDR Arrangement Management API.
```

         order = "-13"
         enable = true
          
         [event_listener.properties]
         TokenEndpointAlias = "https://<APIM_HOST>:8243/arrangements/1.0.0"
        ```
     3. Add the given filters and filter mappings to enforce private key JWT client authentication to the CDR Arrangement Revocation endpoint.
        ```
          [[tomcat.filter]]
          name = "PrivateKeyJwtAuthenticationFilter"
          class = "com.wso2.finance.open.banking.identity.extensions.filter.PrivateKeyJwtAuthenticationFilter"
          
          [[tomcat.filter_mapping]]
          name = "TokenFilter"
          url_pattern = "/arrangements/*"
          
          [[tomcat.filter_mapping]]
          name = "PrivateKeyJwtAuthenticationFilter"
          url_pattern = "/arrangements/*"
        ```
     4. Add the following configuration changes to the Data Recipient's consent revocation endpoint.

        ```
        [open_banking.au.recipient_consent_revocation_endpoint]
        enable = true
        
        [open_banking.uk]
        consent_revocation_listener = "com.wso2.finance.open.banking.uk.consent.mgt.listener.AUConsentRevocationListener"
        ```

>      5. The “sub” and “iss” claims in the Bearer JWT of the request sent to the Data Recipient's CDR Arrangement 

Reviewer: ```suggestion
     5. The `sub` and `iss` claims in the Bearer JWT of the request sent to the Data Recipient's CDR Arrangement 
```

        Revocation endpoint should be the ID of the Data Holder, which was obtained from the CDR Register. 
        ```
        [open_banking.au.data_holder]
        client_id = "dataholderbrand"
        ```

>      6. To persist recipient_base_uri as service provider metadata, add the following configuration to the `<APIM_HOME>/repository/conf/deployment.toml` file. 

Reviewer: ```suggestion
     6. To persist `recipient_base_uri` as service provider metadata, add the following configuration to the `<APIM_HOME>/repository/conf/deployment.toml` file. 
```

      "redirect_uris": [
      "https://www.mockcompany.com.au/redirects/redirect1",
      "https://www.mockcompany.com.au/redirects/redirect2"
      ],
      "sector_identifier_uri": "https://www.mockcompany.com.au/sector_identifier",
      "logo_uri": "https://www.mockcompany.com.au/logos/logo1.png",
      "tos_uri": "https://www.mockcompany.com.au/tos.html",
      "policy_uri": "https://www.mockcompany.com.au/policy.html",
      "jwks_uri": "https://www.mockcompany.com.au/jwks",
      "revocation_uri": "https://www.mockcompany.com.au/revocation",
      "recipient_base_uri": "https://www.mockcompany.com.au",
      "software_id": "740C368F-ECF9-4D29-A2EA-0514A66B0CDE",
      "software_roles": "data-recipient-software-product",
      "scope": "openid profile bank:accounts.basic:read bank:accounts.detail:read bank:transactions:read bank:payees:read bank:regular_payments:read common:customer.basic:read common:customer.detail:read cdr:registration"
      }
      ```

      The Data Recipient's CDR Arrangement Revocation endpoint is `<recipient_base_uri>/arrangements/revoke`. 
      For example, `https://www.mockcompany.com.au/arrangements/revoke`

>       - This needs to be done before enabling this feature, using a DCR PUT request and if the Data Recipients modify this 

Reviewer: ```suggestion
      - This needs to be done before enabling this feature, using a DCR PUT request.  If the Data Recipients modify this 
```

        endpoint, they should update their client registrations with each Data Holder as well.

### Deploying the Arrangement Management API

> 1. Sign in to the API Publisher portal at `https:// <APIM_HOST>:9443/publisher` with `creator/publisher` privileges.

Reviewer: ```suggestion
1. Sign in to the API Publisher portal at `https://<APIM_HOST>:9443/publisher` with `creator/publisher` privileges.
```

   ![business-plan](../assets/img/get-started/quick-start-guide/business-plan.png)

10. Click **Save**.

11. Once you get the message that the API is successfully updated, use the left menu panel and select **API Configurations > Properties**.
    
    ![select_properties](../assets/img/try-out/arrangement-management-api/select-properties.png)  

12. Click  **Add New Property**.
    ![add_new_property](../assets/img/try-out/arrangement-management-api/create-new-property.png)  

13. Add the API Properties according to your API and click the **Add** button. 

       | Property Name | Property Value |
       |----------------|------------- |  
       | ob-spec       | au             |
       | ob-api-type   |  cdr-arrangement|

14. Click **SAVE**.

15. Go back to **Overview** using the left menu panel. 

16. Click **PUBLISH**

17. The published API is available in the Developer Portal at `https://<APIM_HOST>:9443/devportal`.

## Revoke a sharing arrangement

This endpoint is to revoke a sharing arrangement (consent) between the Data Holder and the Data Recipient. This endpoint 

> must be implemented by both Data Holders and Data Recipients and notifies each other when a CDR Arrangement ID is revoked.

Reviewer: ```suggestion
must be implemented by both Data Holders and Data Recipients and notify each other when a CDR Arrangement ID is revoked.
```
